bookkeeper-server/src/test/java/org/apache/bookkeeper/bookie/CompactionTest.java

```diff
@@ -835,6 +837,70 @@ public void checkpointComplete(Checkpoint checkpoint,
         storage.gcThread.doCompactEntryLogs(threshold);
     }
 
+    /**
+     * Test extractMetaFromEntryLogs optimized method to avoid excess memory usage.
+     */
+    @Test(timeout = 60000)
+    public void testExtractMetaFromEntryLogs() throws Exception {
+        tearDown(); // I dont want the test infrastructure
```

Comment:
If you don't want the test infra in this test suite, is it better to just have this as a separate test case?